LabExT/Movement/PathPlanning.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+LabExT  Copyright (C) 2022  ETH Zurich and Polariton Technologies AG
+This program is free software and comes with ABSOLUTELY NO WARRANTY; for details see LICENSE file.
+"""
+
+from abc import ABC, abstractmethod
+from typing import Tuple, Type
+
+import numpy as np
+
+from LabExT.Movement.Transformations import ChipCoordinate
+from LabExT.Movement.config import Orientation
+
+
+class StagePolygon(ABC):
+    """
+    Abstract base class to generate polygons for stages.
+
+    This class cannot be initialised.
+    """
+
+    @abstractmethod
+    def __init__(
+        self,
+        orientation: Orientation,
+        safety_distance: float = 75.0
+    ) -> None:
+        pass
+
+    @abstractmethod
+    def stage_in_meshgrid(
+        self,
+        position: Type[ChipCoordinate],
+        mesh_x: np.ndarray,
+        mesh_y: np.ndarray
+    ) -> np.ndarray:
+        """
+        Returns a mask if a point in the meshgrid is in the stage.
+        """
+        pass
+
+
+class SingleModeFiber(StagePolygon):
+    """
+    Polygon for single mode fiber.
+    """
+
+    FIBER_LENGTH = 8e4  # [um] (8cm)
+    FIBER_RADIUS = 75.0  # [um]
+
+    def __init__(
+        self,
+        orientation: Orientation,
+        safety_distance: float = 75.0
+    ) -> None:
+        """
+        Constructor for new single mode fiber polygon.
+
+        Parameters
+        ----------
+        orientation: Orientation
+            Orientation of the stage in space
+        safety_distance: float
+            Safety distance around the fiber in um.
+        """
+        self.orientation = orientation
+        self.safety_distance = safety_distance
+
+    def stage_in_meshgrid(self,
+                          position: Type[ChipCoordinate],
+                          mesh_x: np.ndarray,
+                          mesh_y: np.ndarray,
+                          ) -> np.ndarray:
+        """
+        Returns a mask if if a point of the single mode fiber in the meshgrid is in the stage.
+
+        Parameters
+        ----------
+        position : ChipCoordinate
+            Current position of the stage in Chip-Coordinates
+        mesh_x : np.ndarray
+            X values of the meshgrid
+        mesh_y : np.ndarray
+            Y values of the meshgrid
+        """
+        x_min, x_max, y_min, y_max = self._create_outline(position)
+        return np.logical_and(
+            np.logical_and(x_min <= mesh_x, mesh_x <= x_max),
+            np.logical_and(y_min <= mesh_y, mesh_y <= y_max))
```

Comment:
If the used gridsize in the meshgrid is too large, it might be that the obstacle completely falls between mesh points, in which case the obstacle is not seen from the potential field algorithm. Please make sure that an obstacle always has at minimum one point assigned.

Also, you could artificially "enlargen" the obstacle by `gridsize/2 + epsilon` s.t. it is guaranteed that we hit meshgrid points?

Reply:
I have changed the following: The method `stage_in_meshgrid` is now also passed the grid size. With this, each obstacle can calculate a polygon that is safely in the grid. For the `SingleModeFiber` I did the following: If, for example, the absolute distance between `x_max` and `x_min` is less than or equal to the grid size, I add half the grid size plus 10 as an epsilon to `x_max`, and subtract this value for `x_min`.

Further, I want to introduce to the path algorithm the following: If the stage -position is in the grid, `stage_in_meshgrid` must return a mask with at least one true-value. When the obstacles are created we can check if this is the case. If not, we throw a RunTime error.

Reply:
Very good suggestion for an addition with the minimum-one-enabled obstacle :)